pkg/reconciler/delivery/delivery.go

```diff
@@ -38,22 +38,28 @@ const (
 	KCDName = "knative-continuous-delivery"
 	// AnnotationKey is the string used in ObjectMeta.Annotations map for any Route object
 	AnnotationKey = "KCDLastRouteUpdate"
+	// RevisionGenerationKey is the label key for querying how "old" a Revision is
+	RevisionGenerationKey = "serving.knative.dev/configurationGeneration"
+	// WaitForReady makes sure that when a newly created Revision becomes ready, it triggers the reconciler
+	WaitForReady = 5 * time.Second
 	// TimeFormat specifies the format used by time.Parse and time.Format
 	TimeFormat = time.RFC3339
 )
 
 // Reconciler implements controller.Reconciler
 type Reconciler struct {
-	client        clientset.Interface
-	routeLister   listers.RouteLister
-	followup      enqueueFunc
-	timeProvider  timeSnapshotFunc
+	client         clientset.Interface
+	routeLister    listers.RouteLister
+	revisionLister listers.RevisionLister
+	followup       enqueueFunc
+	timeProvider   timeSnapshotFunc
```

Comment:
you should probably use k8s standard Clock interface here. It can be mocked with however you need.

Reply:
I will look into it!

Reply:
I deleted the timeProvider altogether because it's not needed anymore. The Clock interface seems pretty neat, and I'll keep it in mind for the future